function-aws-api-gateway-proxy/src/main/java/io/micronaut/aws/function/apigatewayproxy/ApiGatewayProxyRequestEventHandler.java

```diff
@@ -0,0 +1,115 @@
+import io.micronaut.core.execution.ExecutionFlow;
+import io.micronaut.core.execution.ImperativeExecutionFlow;
+import io.micronaut.core.util.CollectionUtils;
+import io.micronaut.function.aws.MicronautRequestHandler;
+import io.micronaut.http.HttpHeaders;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.MutableHttpResponse;
+import io.micronaut.http.server.RouteExecutor;
+import jakarta.inject.Inject;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * {@link com.amazonaws.services.lambda.runtime.RequestHandler(APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent) which starts a Micronaut Application Context in the constructor and handles the full route processing lifecycle for a request.
+ * @author Sergio del Amo
+ * @since 4.0.0
+ *
+ */
+public class ApiGatewayProxyRequestEventHandler extends MicronautRequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+
+    @Inject
+    RouteExecutor routeExecutor;
+
+    @Inject
+    ConversionService conversionService;
+
+    /**
+     * Constructor.
+     */
+    public ApiGatewayProxyRequestEventHandler() {
+    }
+
+    /**
+     * Constructor.
+     * @param applicationContext Application Context
+     */
+    public ApiGatewayProxyRequestEventHandler(ApplicationContext applicationContext) {
+        super(applicationContext);
+    }
+
+    /**
+     * Constructor.
+     * @param applicationContextBuilder ApplicationContextBuilder
+     */
+    public ApiGatewayProxyRequestEventHandler(ApplicationContextBuilder applicationContextBuilder) {
+        super(applicationContextBuilder);
+    }
+
+    @Override
+    public APIGatewayProxyResponseEvent execute(APIGatewayProxyRequestEvent input) {
+        HttpRequest<?> request = new ApiGatewayProxyRequestEventAdapter<>(conversionService, input);
+        HttpResponse<?> response = handle(request);
+        return adaptResponse(request, response);
+    }
+
+    private HttpResponse<?> handle(HttpRequest<?> request) {
+        AmazonApiGatewayRequestLifecycle requestLifecycle = new AmazonApiGatewayRequestLifecycle(routeExecutor, request);
+        ExecutionFlow<MutableHttpResponse<?>> flow = requestLifecycle.run();
+        ImperativeExecutionFlow<MutableHttpResponse<?>> mutableHttpResponseImperativeExecutionFlow = flow.tryComplete();
+        return mutableHttpResponseImperativeExecutionFlow.getValue();
+    }
+
+    private APIGatewayProxyResponseEvent adaptResponse(HttpRequest<?> request, HttpResponse<?> response) {
+        //TODO Extract this adaptation to a  class where we can reuse it in the TCK logic
+        APIGatewayProxyResponseEvent responseEvent = new APIGatewayProxyResponseEvent();
+        response.getBody(String.class).ifPresent(responseEvent::setBody);
+        responseEvent.setStatusCode(response.code());
+
+        // NO idea why conten type is not in the response headers
```

Comment:
My findings led me to [here](https://github.com/micronaut-projects/micronaut-core/blob/4.0.x/http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java#L300-L303)

IMHO, this needs to be ported to something like `http-server-function`